tasks/kernel_matrix_testing/compiler.py

```diff
@@ -84,6 +93,15 @@ def ensure_running(self):
             except Exception as e:
                 raise Exit(f"Failed to start compiler for {self.arch}: {e}") from e
 
+    def ensure_version(self):
+        if not self.is_running:
+            return  # Nothing to do if the container is not running
+
+        image_used = get_docker_image_name(self.ctx, self.name)
+        if image_used != self.image:
+            warn(f"[!] Running compiler image {image_used} is different from the expected {self.image}, will restart")
+            self.start()
```

Comment:
I think `start` needs to remove the `-a` argument from `docker ps` or it might detect a stopped container.

Reply:
fixed!

Reply:
I _think_ if you have a stopped container, `start` still won't work because it won't try to remove the stopped container and docker won't let you have two containers with the same name.

Reply:
We may need to differentiate between a running container and a stopped container in the code to make this clear.